RealmSwift-swift2.2/Tests/RealmTests.swift

```diff
@@ -255,6 +255,29 @@ class RealmTests: TestCase {
         }
         XCTAssertEqual(try! Realm().objects(SwiftStringObject.self).count, 0)
     }
+    
+    func testThrowsWrite() {
+        enum TestError: ErrorType { case Exception }
+        
+        do {
+            try Realm().write {
+                throw TestError.Exception
+            }
+            XCTFail("Expected exception thrown in write block to be rethrown.")
+        }
+        catch TestError.Exception { }
```

Comment:
`assertThrows` only checks for Objective-C exceptions. I don't think there's any correspondence between `try` and Obj-C exceptions, so you may need to write your own Swift error-specific test harness.